File: LeetCode/Python3/LeetCode0055_JumpGame.py
# LeetCode0055_JumpGame.py
# https://leetcode.com/problems/jump-game/
class Solution:
	def canJump( self, nums: List[int] ) -> bool:
		numsLen = len( nums )
		canQ = [ True ] + [ False ] * (numsLen - 1)

		for idx, n in enumerate( nums ):
			if canQ[idx]:
				for i in range( n, 0, -1 ):
					if (idx + i < numsLen and not canQ[idx + i]):
						canQ[idx + i] = True
						if idx + i == numsLen - 1:
							return True
					elif ( idx + i < numsLen and canQ[idx + i] == True ):
						break

		return canQ[numsLen - 1]

File: LeetCode/C++/LeetCode0055_JumpGame.cpp
// LeetCode0055_JumpGame.cpp
// https://leetcode.com/problems/jump-game/
class Solution {
public:
	bool canJump( vector<int>& nums ) {
		bool canCome[30000] = { false, };

		canCome[0] = true;
		for ( int i = 0; i < nums.size() - 1; i++ ) {
			if ( canCome[i] ) {
				for ( int j = nums[i]; j >= 1; j-- ) {
					int next = j + i;
					if ( next < nums.size() && !canCome[next] )
						canCome[next] = true;
					else if ( canCome[next] == true )
						break;
				}
			} else
				break;
		}

		return canCome[nums.size() - 1];
	}
};